crypto/ec/ec_mult.c

```diff
@@ -101,6 +101,227 @@ void EC_ec_pre_comp_free(EC_PRE_COMP *pre)
     OPENSSL_free(pre);
 }
 
+#define EC_POINT_BN_set_flags(P, flags) do { \
+    BN_set_flags((P)->X, (flags)); \
+    BN_set_flags((P)->Y, (flags)); \
+    BN_set_flags((P)->Z, (flags)); \
+} while(0)
+
+/*-
+ * This functions computes (in constant time) a point multiplication over the
+ * EC group.
+ *
+ * At a high level, it is Montgomery ladder with conditional swaps.
+ *
+ * It performs either a fixed scalar point multiplication
+ *          (scalar * generator)
+ * when point is NULL, or a generic scalar point multiplication
+ *          (scalar * point)
+ * when point is not NULL.
+ *
+ * scalar should be in the range [0,n) otherwise all constant time bets are off.
+ *
+ * NB: This says nothing about EC_POINT_add and EC_POINT_dbl,
+ * which of course are not constant time themselves.
+ *
+ * The product is stored in r.
+ *
+ * Returns 1 on success, 0 otherwise.
+ */
+static int ec_mul_consttime(const EC_GROUP *group, EC_POINT *r,
+                            const BIGNUM *scalar, const EC_POINT *point,
+                            BN_CTX *ctx)
+{
+    int i, order_bits, group_top, kbit, pbit, Z_is_one;
+    EC_POINT *s = NULL;
+    BIGNUM *k = NULL;
+    BIGNUM *lambda = NULL;
+    BN_CTX *new_ctx = NULL;
+    int ret = 0;
+
+    if (ctx == NULL && (ctx = new_ctx = BN_CTX_secure_new()) == NULL)
+        goto err;
+
+    if ((group->order == NULL) || (group->field == NULL))
+        goto err;
```

Comment:
Every time I see such construct I struggle with question "why". Can there be an EC_GROUP instance without field and order? Or is NULL possible only as result of memory corruption? Trouble is that if so, then wouldn't it be as appropriate to proceed without check and crash? Thing is that masking would take you further from root cause and only complicate trouble-shooting. But even trouble-shooting aside, if there is corruption you are genuinely interested to take as little as possible security-critical decision, ideally none, or in other words crash as soon as possible.

As for possibility of EC_GROUP without field and order. Even in extreme case when EC_GROUP is instantiated from user-supplied ASN.1 structure it, EC_GROUP_new_from_ecparameters, refuses to create EC_GROUP without field and order (and of course bunch of other fields).

To summarize, suggestion is to remove this condition. As well as check for group->generator below. As for other `goto err`-s. Question is whether or not it would be appropriate to set some ECerrs... I mean when application fails what would be user's chances to figure out the cause... Yes, most of callees set their own errors, but question is if they would be too generic to trouble-shoot...